kloppy/infra/serializers/tracking/statsperform.py

```diff
@@ -0,0 +1,271 @@
+    def __init__(
+        self,
+        limit: Optional[int] = None,
+        sample_rate: Optional[float] = None,
+        coordinate_system: Optional[Union[str, Provider]] = None,
+        only_alive: Optional[bool] = True,
+    ):
+        super().__init__(limit, sample_rate, coordinate_system)
+        self.only_alive = only_alive
+
+    @property
+    def provider(self) -> Provider:
+        return Provider.STATSPERFORM
+
+    @classmethod
+    def __get_periods(cls, tracking):
+        """Gets the Periods contained in the tracking data."""
+        lines = tracking.decode("ascii").splitlines()
+        period_ids = []
+        frame_ids = []
+
+        for line in lines:
+            time_info = line.split(";")[1].split(",")
+            period_ids.append(time_info[1])
+            frame_ids.append(time_info[0])
+
+        unique_period_ids = list(dict.fromkeys(period_ids).keys())
+        periods = []
+
+        for period_id in unique_period_ids:
+            period_start_index = period_ids.index(period_id)
+            period_end_index = (
+                len(period_ids) - period_ids[::-1].index(period_id) - 1
+            )
+            periods.append(
+                Period(
+                    id=int(period_id),
+                    start_timestamp=int(frame_ids[period_start_index]),
+                    end_timestamp=int(frame_ids[period_end_index]),
+                )
+            )
+
+        return periods
+
+    @classmethod
+    def __get_frame_rate(cls, tracking):
+        """gets the frame rate of the tracking data"""
+        lines = tracking.decode("ascii").splitlines()
+        timestamps = []
+        num_frames = 5
+
+        for i in [0, num_frames]:
+            timestamp = int(lines[i].split(";")[1].split(",")[0]) / 1000
+            timestamps.append(timestamp)
+
+        duration = timestamps[1] - timestamps[0]
+        frame_rate = num_frames / duration
+
+        return frame_rate
+
+    @classmethod
+    def _frame_from_framedata(cls, teams, period, frame_data):
+        components = frame_data[1].split(":")
+        frame_info = components[0].split(";")
+
+        frame_id = int(frame_info[0])
+        frame_timestamp = int(frame_info[1].split(",")[0]) / 1000
+        match_status = int(frame_info[1].split(",")[2])
+
+        ball_state = BallState.ALIVE if match_status == 0 else BallState.DEAD
+        ball_owning_team = None
+
+        if len(components) > 2:
+            ball_x, ball_y, ball_z = map(
+                float, components[2].split(";")[0].split(",")
+            )
+            ball_coordinates = Point3D(ball_x, ball_y, ball_z)
+        else:
+            ball_coordinates = None
+
+        players_data = {}
+        player_info = components[1]
+        for player_data in player_info.split(";")[:-1]:
+            (
+                team_side_id,
+                player_id,
+                jersey_no,
+                x,
+                y,
+            ) = player_data.split(",")
```

Comment:
I would do all the required type casts immediately after the `player_data.split(",")` call to avoid cluttering the "business logic" with type casts.